### PR TITLE
feat!: making the settings store search more flexible and adding support for conditional put - NOTE: changes the API contract

### PR DESCRIPTION
**Pull request by klcodanr**
Please ensure your pull request adheres to the following guidelines:
- [ ] make sure to link the related issues in this description
- [ ] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues

Thanks for contributing!

### CODE REVIEW

File: src/request-handler.js [hunk 1]
```diff
@@ -34,12 +35,26 @@ import { logger } from '@adobe/helix-universal-logger';
...
+  /**
+   * An OAuth authentication event, only supported for handlers with OAuth support
+   */
   static ACTION_AUTHENTICATE = 'authenticate';
 
+  /**
+   * An update event from the source, must be supported
+   */
+  static ACTION_UPDATE = 'update';
+
```
Comment (jdelbick) [inline src/request-handler.js:57]: todo: still need to move this class to commons

File: src/settings.js [hunk 2]
```diff
@@ -70,6 +70,19 @@ export class SettingsStore {
...
 
+  /**
+   *
+   * @param {Record<string, string>} item the item to serialize
+   * @returns {Record<string,AttributeValue>} the serialized item
+   */
+  static serializeItem(item) {
+    const newItem = {};
+    Object.keys(item).forEach((k) => {
+      newItem[k] = { S: item[k] };
```
Comment (jdelbick) [inline src/settings.js:81]: what is `S`?
what about using `clone`?

**Review by jdelbick — APPROVED** [review]
nitpick question about why setting store has the `S` source key. Suggestion to give it a more readable name if its not something that is hardcoded

**Review by jdelbick — COMMENTED** [review]
quick qs, does this break the current behavior of requestHandler? (it looks like it just adds methods for that class but want to confirm since we use this in a lot of other lambdas now

**Comment by klcodanr** [comment]
No the breaking change is to the settings class.

File: src/settings.js (hunk 2 above)
Comment (klcodanr) [reply src/settings.js:81]: It’s a super jenky construct used for typing in the DynamoDB SDK. It’s not the world’s friendlies SDK to work with

**Pull request merged**